includes/polling/core.inc.php

```diff
@@ -58,19 +58,12 @@
     $device['uptime']       = $uptime;
 }//end if
 
-$poll_device['sysLocation'] = str_replace('"', '', $poll_device['sysLocation']);
-
-// Rewrite sysLocation if there is a mapping array (database too?)
-if (!empty($poll_device['sysLocation']) && (is_array(Config::get('location_map')) || is_array(Config::get('location_map_regex')) || is_array(Config::get('location_map_regex_sub')))) {
-    $poll_device['sysLocation'] = rewrite_location($poll_device['sysLocation']);
-}
+set_device_location($poll_device['sysLocation'], $device, $update_array);
 
 $poll_device['sysContact'] = str_replace('"', '', $poll_device['sysContact']);
 
-foreach (array('sysLocation', 'sysContact') as $elem) {
-    if ($poll_device[$elem] == 'not set') {
-        $poll_device[$elem] = '';
-    }
+if ($poll_device['sysContact'] == 'not set') {
```

Comment:
this looks like a device specific default setting,
not so good to handle this in LibreNMS